src/Widgets/ListSettingsPopover.vala

```diff
@@ -234,4 +219,83 @@ public class Tasks.ListSettingsPopover : Gtk.Popover {
 
         Application.settings.bind ("show-completed", show_completed_switch, "active", GLib.SettingsBindFlags.DEFAULT);
     }
+
+    private void select_task_list_color (E.Source source) {
+        switch (get_task_list_color (source)) {
+            case "#c6262e":
+                color_button_red.active = true;
+                break;
+            case "#f37329":
+                color_button_orange.active = true;
+                break;
+            case "#e6a92a":
+                color_button_yellow.active = true;
+                break;
+            case "#68b723":
+                color_button_green.active = true;
+                break;
+            case "#0e9a83":
+                color_button_mint.active = true;
+                break;
+            case "#3689e6":
+                color_button_blue.active = true;
+                break;
+            case "#a56de2":
+                color_button_purple.active = true;
+                break;
+            case "#de3e80":
+                color_button_pink.active = true;
+                break;
+            case "#8a715e":
+                color_button_brown.active = true;
+                break;
+            case "#667885":
+                color_button_slate.active = true;
+                break;
+            default:
+                color_button_none.active = true;
+                break;
+        }
+    }
+
+    private string get_task_list_color (E.Source source) {
+        if (source.has_extension (E.SOURCE_EXTENSION_TASK_LIST)) {
+            var task_list = (E.SourceTaskList) source.get_extension (E.SOURCE_EXTENSION_TASK_LIST);
+            return task_list.dup_color ();
+        }
+        return "";
+    }
+
+    private void update_task_list_color (E.Source source, string color) {
+        if (get_task_list_color (source) == color) {
+            return;
+        }
+
+        Tasks.Application.model.update_task_list_color.begin (source, color, (obj, res) => {
+            try {
+                Tasks.Application.model.update_task_list_color.end (res);
+            } catch (Error e) {
+                select_task_list_color (source);
+                dialog_update_task_list_color_error (e);
+            }
+        });
+    }
+
+    private void dialog_update_task_list_color_error (Error e) {
+        string error_message = e.message;
+
+        GLib.Idle.add (() => {
```

Comment:
I don't see that we're using threads here, so it doesn't seem like an idle is necessary

Reply:
Interesting: It works without `GLib.Idle`...

Does this mean `GLib.Idle` is not needed for async operations? Asking because coming from iOS, there we also need to dispatch UI work explicitly to the main thread in case work was done asynchronously.

Reply:
@tintou would be able to explain this better, but I think what you're talking about is wrapping something in a [GLib.MainLoop](https://valadoc.org/glib-2.0/GLib.MainLoop.html) which afaik you only need to do if you're using [GLib.Thread](https://valadoc.org/glib-2.0/GLib.Thread.html)